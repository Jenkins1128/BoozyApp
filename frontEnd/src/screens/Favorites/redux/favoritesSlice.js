import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
