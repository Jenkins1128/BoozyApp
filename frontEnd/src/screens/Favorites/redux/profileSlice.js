import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
