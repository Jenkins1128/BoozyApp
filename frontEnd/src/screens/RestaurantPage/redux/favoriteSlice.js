import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const favoriteAsync = createAsyncThunk('favoriteAsync/status', async (data, { rejectWithValue }) => {
	const { restaurantId, name } = data;
	try {
		//favorite restaurant
		await axios.post(`${baseUrl}/user/add`, {
			restaurantId: restaurantId,
			name: name
		});
		//add restaurants to your favorites
		const response = await axios.post(`${baseUrl}/favorites`, {
			restaurantId: restaurantId,
			name: name
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const favoriteSlice = createSlice({
	name: 'favoriteAsync',
	initialState,
	reducers: {
		resetFavorite: (state) => {
			state.currentState.favoriteRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(favoriteAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					favoriteColor: payload[0]['contains'] ? 'red' : 'white',
					favoriteRequestStatus: 'fulfilled'
				});
			})
			.addCase(favoriteAsync.rejected, (state) => {
				state.currentState.favoriteRequestStatus = 'rejected';
			});
	}
});

export const { resetFavorite } = favoriteSlice.actions;

export const selectFavoriteState = (state) => state.favorite.currentState;
export default favoriteSlice.reducer;
