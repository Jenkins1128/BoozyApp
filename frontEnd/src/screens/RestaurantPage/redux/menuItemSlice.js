import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const postMenuItemAsync = createAsyncThunk('postMenuItemAsync/status', async (data, { rejectWithValue }) => {
	const { description, price, restaurantName, restaurantId } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/menu', {
			content: description,
			price: price,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const postMenuItemSlice = createSlice({
	name: 'postMenuItemAsync',
	initialState,
	reducers: {
		updatePrice: (state, { payload }) => {
			state.currentState = { ...state.currentState, price: payload.price };
		},
		updateDescription: (state, { payload }) => {
			state.currentState = { ...state.currentState, description: payload.description };
		},
		resetMenuItem: (state) => {
			state.currentState = { ...state.currentState, price: '', description: '', postMenuItemRequestStatus: 'idle' };
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(postMenuItemAsync.fulfilled, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'fulfilled' };
			})
			.addCase(postMenuItemAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, postMenuItemRequestStatus: 'rejected' };
			});
	}
});

export const { updatePrice, updateDescription, resetMenuItem } = postMenuItemSlice.actions;

export const selectMenuItemState = (state) => state.menuItem.currentState;
export default postMenuItemSlice.reducer;
