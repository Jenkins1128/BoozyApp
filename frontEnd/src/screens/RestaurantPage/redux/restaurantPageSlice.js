import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		restaurantId: '',
		restaurantName: '',
		restaurantImage: 'https://boozyimage.s3.us-east-2.amazonaws.com/logo.png',
		phoneNumber: '',
		allCategories: '',
		rating: 0.0,
		reviews: 0,
		favoriteColor: 'white',
		changeFavoritedColor: '',
		starCount: 0,
		favorited: false,
		rated: false,
		price: 0.0,
		description: '',
		menuItemArray: []
	}
};

export const getMenuItemsAsync = createAsyncThunk('getMenuItemsAsync/status', async (restaurantId, { rejectWithValue }) => {
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/menu/list', {
			restaurantId: restaurantId
		});
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const restaurantPageSlice = createSlice({
	name: 'getMenuItemsAsync',
	initialState,
	reducers: {
		updateState: (state, { payload }) => {
			console.log('payload', payload);
			state.currentState = {
				...state.currentState,
				restaurantId: payload.params.id,
				restaurantName: payload.params.name,
				restaurantImage: payload.params.img,
				phoneNumber: payload.params.phone,
				allCategories: payload.params.allCategories,
				rating: payload.params.rating,
				reviews: payload.params.reviews,
				rated: payload.params.alreadyRated,
				favorited: payload.params.alreadyFavorited,
				starCount: 0
			};
		},
		updateFavoriteColor: (state, { payload }) => {
			state.currentState = {
				...state.currentState,
				favoriteColor: payload.color
			};
		},
		updateStarCount: (state, { payload }) => {
			state.currentState = {
				...state.currentState,
				starCount: payload.starCount
			};
		},
		updateMenuItems: (state, { payload }) => {
			state.currentState = { ...state.currentState, menuItemArray: [...state.currentState.menuItemArray, payload.menuItem] };
		},
		resetRestaurantPageRequestStatus: (state) => {
			state.currentState.restaurantPageRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getMenuItemsAsync.fulfilled, (state, { payload }) => {
				state.currentState = { ...state.currentState, restaurantPageRequestStatus: 'fulfilled', menuItemArray: payload };
			})
			.addCase(getMenuItemsAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, restaurantPageRequestStatus: 'rejected' };
			});
	}
});

export const { updateState, updateFavoriteColor, updateStarCount, updateMenuItems, resetRestaurantPageRequestStatus } = restaurantPageSlice.actions;

export const selectRestaurantPageState = (state) => state.restaurantPage.currentState;
export default restaurantPageSlice.reducer;
