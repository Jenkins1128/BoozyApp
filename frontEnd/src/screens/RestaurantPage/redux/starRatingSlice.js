import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const starRatingAsync = createAsyncThunk('starRatingAsync/status', async (data, { rejectWithValue }) => {
	const { rating, restaurantId, restaurantName } = data;
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/user/' + restaurantId + '/restaurants', {
			content: rating,
			name: restaurantName
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const starRatingSlice = createSlice({
	name: 'starRatingAsync',
	initialState,
	reducers: {
		resetStarRatingRequestStatus: (state) => {
			state.currentState.starRatingRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(starRatingAsync.fulfilled, (state) => {
				state.currentState.starRatingRequestStatus = 'fulfilled';
			})
			.addCase(starRatingAsync.rejected, (state) => {
				state.currentState.starRatingRequestStatus = 'rejected';
			});
	}
});

export const { resetStarRatingRequestStatus } = starRatingSlice.actions;

export const selectStarRatingState = (state) => state.starRating.currentState;
export default starRatingSlice.reducer;
