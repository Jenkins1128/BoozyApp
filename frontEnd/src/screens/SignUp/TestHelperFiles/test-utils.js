import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
