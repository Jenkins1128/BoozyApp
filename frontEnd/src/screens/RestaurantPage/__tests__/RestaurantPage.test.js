import React from 'react';
import RestaurantPage from '../RestaurantPage';
import { reduxRender, cleanup } from '../../../TestHelperFiles/redux/test-utils';
import RestaurantPageFuncs from '../../../TestHelperFiles/Functions/RestaurantPageFuncs';
import favoriteReducer, { favoriteAsync } from '../redux/favoriteSlice';
import menuItemReducer, { postMenuItemAsync } from '../redux/menuItemSlice';
import restaurantPageReducer, { getMenuItemsAsync } from '../redux/restaurantPageSlice';
import starRatingReducer, { starRatingAsync } from '../redux/starRatingSlice';
import axios from 'axios';
import { baseUrl, defaultRestaurantImageUrl } from '../../../helpers/constants';
import { configureStore } from '@reduxjs/toolkit';

const createTestProps = (props) => ({
	navigation: {
		navigate: jest.fn(),
		addListener: jest.fn()
	},
	...props
});

describe('<RestaurantPage />', () => {
	let rendered;

	beforeEach(() => {
		props = createTestProps({});
		rendered = reduxRender(<RestaurantPage {...props} />);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should wrap view with a flexible wrapper', () => {
		const keyboardAvoidingViewComponent = rendered.getByTestId('container');
		const given = keyboardAvoidingViewComponent.props.style;
		//padding-bottom 0 added from KeyboardAvoidingView behavior prop
		const result = [{ flex: 1 }, { paddingBottom: 0 }];
		expect(result).toMatchObject(given);
	});

	it('should render inner styles and center items correctly', () => {
		const innerViewComponent = rendered.getByTestId('inner');
		const given = innerViewComponent.props.style;
		//padding-bottom 0 added from KeyboardAvoidingView behavior prop
		const result = {
			flex: 1,
			justifyContent: 'center',
			alignItems: 'center'
		};
		expect(result).toMatchObject(given);
	});

	//functions
	describe('showErrorAlert', () => {
		it('calls sideEffect', () => {
			jest.spyOn(RestaurantPageFuncs, 'showErrorAlert');
			RestaurantPageFuncs.showErrorAlert('Network failed...');
			expect(RestaurantPageFuncs.showErrorAlert).toHaveBeenCalledTimes(1);
			expect(RestaurantPageFuncs.showErrorAlert).toHaveBeenCalledWith('Network failed...');
		});
	});

	describe('getRestaurantDataOnFocus', () => {
		it('calls sideEffect', () => {
			jest.spyOn(RestaurantPageFuncs, 'getRestaurantDataOnFocus');
			RestaurantPageFuncs.getRestaurantDataOnFocus();
			expect(RestaurantPageFuncs.getRestaurantDataOnFocus).toHaveBeenCalledTimes(1);
		});
	});

	describe('showMenuItemOverlay', () => {
		it('calls sideEffect', () => {
			jest.spyOn(RestaurantPageFuncs, 'showMenuItemOverlay');
			RestaurantPageFuncs.showMenuItemOverlay(true);
			expect(RestaurantPageFuncs.showMenuItemOverlay).toHaveBeenCalledTimes(1);
		});
	});

	describe('changeColor', () => {
		it('calls sideEffect', () => {
			jest.spyOn(RestaurantPageFuncs, 'changeColor');
			RestaurantPageFuncs.changeColor('white');
			expect(RestaurantPageFuncs.changeColor).toHaveBeenCalledTimes(1);
		});
	});

	describe('onStarRatingPressed', () => {
		it('calls sideEffect', () => {
			jest.spyOn(RestaurantPageFuncs, 'onStarRatingPressed');
			RestaurantPageFuncs.onStarRatingPressed(4);
			expect(RestaurantPageFuncs.onStarRatingPressed).toHaveBeenCalledTimes(1);
		});
	});

	describe('getDataFromMenuItem', () => {
		it('calls sideEffect', () => {
			jest.spyOn(RestaurantPageFuncs, 'getDataFromMenuItem');
			RestaurantPageFuncs.getDataFromMenuItem();
			expect(RestaurantPageFuncs.getDataFromMenuItem).toHaveBeenCalledTimes(1);
		});
	});

	describe('menuItemSubmitted', () => {
		it('calls sideEffect', () => {
			jest.spyOn(RestaurantPageFuncs, 'menuItemSubmitted');
			RestaurantPageFuncs.menuItemSubmitted();
			expect(RestaurantPageFuncs.menuItemSubmitted).toHaveBeenCalledTimes(1);
		});
	});

	describe('favorite', () => {
		it('calls sideEffect', () => {
			jest.spyOn(RestaurantPageFuncs, 'favorite');
			RestaurantPageFuncs.favorite();
			expect(RestaurantPageFuncs.favorite).toHaveBeenCalledTimes(1);
		});
	});

	describe('dismiss', () => {
		it('calls sideEffect', () => {
			jest.spyOn(RestaurantPageFuncs, 'dismiss');
			RestaurantPageFuncs.dismiss();
			expect(RestaurantPageFuncs.dismiss).toHaveBeenCalledTimes(1);
		});
	});

	//redux integrations
	describe('redux', () => {
		it('should dispatch favoriteAsync and state updated fulfilled', async () => {
			const restaurant = {
				restaurantId: 0,
				name: 'Texas Roadhouse'
			};
			const postSpy1 = jest.spyOn(axios, 'post').mockResolvedValueOnce(204);
			const postSpy2 = jest.spyOn(axios, 'post').mockResolvedValueOnce([]);
			const store = configureStore({
				reducer: {
					favorite: favoriteReducer
				}
			});
			await store.dispatch(favoriteAsync(restaurant));
			expect(postSpy1).toBeCalledWith(`${baseUrl}/user/add`, restaurant);
			expect(postSpy2).toBeCalledWith(`${baseUrl}/favorites`, restaurant);
			const state = store.getState();
			expect(state).toEqual({
				favorite: {
					currentState: {
						favoriteColor: 'white',
						favoriteRequestStatus: 'fulfilled'
					}
				}
			});
		});

		it('should dispatch postMenuItemAsync and state updated fulfilled', async () => {
			const restaurant = {
				description: 'Pulled pork, fantastic',
				price: 6,
				restaurantName: 'Texas Roadhouse',
				restaurantId: 0
			};
			const postSpy = jest.spyOn(axios, 'post').mockResolvedValueOnce(204);
			const store = configureStore({
				reducer: {
					menuItem: menuItemReducer
				}
			});
			await store.dispatch(postMenuItemAsync(restaurant));
			expect(postSpy).toBeCalledWith(`${baseUrl}/user/${restaurant.restaurantId}/menu`, {
				content: 'Pulled pork, fantastic',
				price: 6,
				name: 'Texas Roadhouse'
			});
			const state = store.getState();
			expect(state).toEqual({
				menuItem: {
					currentState: {
						postMenuItemRequestStatus: 'fulfilled'
					}
				}
			});
		});

		it('should dispatch getMenuItemsAsync and state updated fulfilled', async () => {
			let restaurantId = 0;
			const postSpy = jest.spyOn(axios, 'post').mockResolvedValueOnce([]);
			const store = configureStore({
				reducer: {
					restaurantPage: restaurantPageReducer
				}
			});
			await store.dispatch(getMenuItemsAsync(restaurantId));
			expect(postSpy).toBeCalledWith(`${baseUrl}/menu/list`, {
				restaurantId: restaurantId
			});
			const state = store.getState();
			expect(state).toEqual({
				restaurantPage: {
					currentState: {
						restaurantPageRequestStatus: 'fulfilled',
						restaurantId: '',
						restaurantName: '',
						restaurantImage: defaultRestaurantImageUrl,
						phoneNumber: '',
						allCategories: '',
						rating: 0.0,
						reviews: 0,
						favoriteColor: 'white',
						changeFavoritedColor: '',
						starCount: 0,
						favorited: false,
						rated: false,
						price: 0.0,
						description: '',
						menuItemArray: undefined
					}
				}
			});
		});

		it('should dispatch starRatingAsync and state updated fulfilled', async () => {
			const restaurant = {
				rating: 4,
				restaurantId: 0,
				restaurantName: 'Texas Roadhouse'
			};
			const postSpy = jest.spyOn(axios, 'post').mockResolvedValueOnce(204);
			const store = configureStore({
				reducer: {
					starRating: starRatingReducer
				}
			});
			await store.dispatch(starRatingAsync(restaurant));
			expect(postSpy).toBeCalledWith(`${baseUrl}/user/${restaurant.restaurantId}/restaurants`, {
				content: restaurant.rating,
				name: restaurant.restaurantName
			});
			const state = store.getState();
			expect(state).toEqual({
				starRating: {
					currentState: {
						starRatingRequestStatus: 'fulfilled'
					}
				}
			});
		});
	});
});
