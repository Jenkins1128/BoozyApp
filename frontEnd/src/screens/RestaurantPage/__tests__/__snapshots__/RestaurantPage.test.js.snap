// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantPage /> should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="inner"
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#EB8873",
          "flex": 1,
          "justifyContent": "center",
          "width": "100%",
        }
      }
      testID="header"
    >
      <Image
        source={
          Object {
            "uri": "https://boozyimage.s3.us-east-2.amazonaws.com/logo.png",
          }
        }
        style={
          Object {
            "flex": 2,
            "height": "100%",
            "width": "100%",
          }
        }
        testID="logoImage"
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
          }
        }
        testID="titleHeader"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }
          testID="restaurantName"
        >
          
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "margin": 10,
              "marginBottom": 0,
              "opacity": 1,
            }
          }
          testID="favoriteButton"
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 15,
          }
        }
        testID="titleBody"
      >
        <View
          style={
            Object {
              "flex": 1,
              "marginBottom": 0,
            }
          }
          testID="infoContainer"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 14,
                "marginBottom": 0,
              }
            }
            testID="categoriesText"
          >
            
             • 
            0
            ★
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 14,
                "marginBottom": 0,
              }
            }
            testID="phoneNumberText"
          >
            
             • 
            0
             Yelp Reviews
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "paddingBottom": 0,
            }
          }
          testID="starRatingContainer"
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "width": "100%",
        }
      }
      testID="scrollContainer"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
        testID="menuItemsTitleContainer"
      >
        <Text
          style={
            Object {
              "color": "lightgrey",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginTop": 0,
              "padding": 10,
            }
          }
          testID="menuTitle"
        >
          Menu items
        </Text>
        <View
          style={
            Object {
              "marginRight": 15,
            }
          }
          testID="menuButtonContainer"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#EB8873",
                "borderRadius": 20,
                "opacity": 1,
                "padding": 7,
                "width": 110,
              }
            }
            testID="menuButton"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontWeight": "bold",
                }
              }
              testID="menuItemText"
            >
              + Menu Item
            </Text>
          </View>
        </View>
      </View>
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 30,
          "padding": 20,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateY": 1000,
            },
          ],
        }
      }
      testID="subView"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 20,
          }
        }
        testID="addMenuItemHeader"
      >
        <View
          style={
            Object {
              "marginRight": 20,
            }
          }
          testID="addMenuItemTitleContainer"
        >
          <Text
            style={
              Object {
                "color": "#EB8873",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
            testID="addMenuItemTitle"
          >
            Add Menu Item
          </Text>
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 20,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="cancelButton"
          >
            <Text
              style={
                Object {
                  "color": "#EB8873",
                  "fontWeight": "bold",
                }
              }
              testID="cancelText"
            >
              Cancel
            </Text>
          </View>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        multiline={false}
        onChangeText={[Function]}
        placeholder="Enter Price"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Object {
            "fontSize": 18,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
        testID="priceInput"
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        multiline={false}
        onChangeText={[Function]}
        placeholder="Enter menu item"
        rejectResponderTermination={true}
        returnKeyType="next"
        style={
          Object {
            "fontSize": 18,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
        testID="descriptionInput"
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EB8873",
            "borderRadius": 20,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
          }
        }
        testID="addMenuItemButton"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
          testID="addMenuItemText"
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;
